fence/agents/gce/fence_gce.py

```diff
@@ -3,55 +3,57 @@
 import atexit
 import sys
 sys.path.append("@FENCEAGENTSLIBDIR@")
-from googleapiclient import discovery
-from oauth2client.client import GoogleCredentials
 
-from fencing import *
-from fencing import fail, fail_usage, EC_TIMED_OUT, run_delay
+import googleapiclient
+import oauth2client
+from fencing import fail_usage, run_delay, all_opt, atexit_handler, check_input, process_input, show_docs, fence_action
+
+def translate_status(instance_status):
+	"Returns on | off | unknown."
+	if instance_status == "RUNNING":
+		return "on"
+	elif instance_status == "TERMINATED":
+		return "off"
+	return "unknown"
 
 
 def get_nodes_list(conn, options):
 	result = {}
 	try:
 		instanceList = conn.instances().list(project=options["--project"], zone=options["--zone"]).execute()
 		for instance in instanceList["items"]:
-			status = "unknown"
-			if instance["status"] == "RUNNING":
-				status = "on"
-			elif instance["status"] == "TERMINATED":
-				status = "off"
-			result[instance["id"]] = (instance["name"], status)
-	# TODO: check which Exceptions it can throw
-	except:
-		fail_usage("Failed: Unable to connect to GCE. Check your configuration.")
+			result[instance["id"]] = (instance["name"], translate_status(instance["status"]))
+	except Exception as err:
+		fail_usage("fence_gce.get_nodes_list: {}".format(str(err)))
```

Comment:
I think these fail_usage() calls should still say Failed: and fence_gce would be provided by the logging facility, so e.g. ""Failed: get_nodes_list: {}" in this case.